src/contexts/analytics/AnalyticsContext.tsx

```diff
@@ -7,7 +7,7 @@ import noop from '~/utils/noop';
 
 type EventProps = Record<string, unknown>;
 
-type TrackFn = (eventName: string, eventProps?: EventProps) => void;
+type TrackFn = (eventName: string, eventProps?: EventProps, checkAuth?: boolean) => void;
```

Comment:
Added new props here. 

Lmk if we wanna convert this props to `Object` instead of multiple parameters. If yes, need to refactor all the `useTrack` usage

Reply:
im ok with this for now!